homeassistant/components/samsungtv/config_flow.py

```diff
@@ -95,81 +156,79 @@ async def async_step_import(self, user_input=None):
+                )
+            except socket.gaierror as err:
+                raise data_entry_flow.AbortFlow(RESULT_UNKNOWN_HOST) from err
+            self._name = user_input[CONF_NAME]
             self._title = self._name
 
-            result = await self.hass.async_add_executor_job(self._try_connect)
+            await self.hass.async_add_executor_job(self._try_connect)
+
+            await self._async_set_device_unique_id(raise_on_progress=False)
 
-            if result != RESULT_SUCCESS:
-                return self.async_abort(reason=result)
             return self._get_entry()
 
         return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
 
-    async def async_step_ssdp(self, discovery_info):
-        """Handle a flow initialized by discovery."""
-        host = urlparse(discovery_info[ATTR_SSDP_LOCATION]).hostname
-        ip_address = await self.hass.async_add_executor_job(_get_ip, host)
+    async def async_step_ssdp(self, discovery_info: DiscoveryInfoType):
+        """Handle a flow initialized by ssdp discovery."""
+        self._host = urlparse(discovery_info[ATTR_SSDP_LOCATION]).hostname
+
+        LOGGER.debug("Found Samsung device via ssdp at %s", self._host)
+        await self._async_set_device_unique_id()
 
-        self._host = host
-        self._ip = self.context[CONF_IP_ADDRESS] = ip_address
         self._manufacturer = discovery_info.get(ATTR_UPNP_MANUFACTURER)
-        self._model = discovery_info.get(ATTR_UPNP_MODEL_NAME)
-        self._name = f"Samsung {self._model}"
-        self._id = discovery_info.get(ATTR_UPNP_UDN)
+        if not self._model:
+            self._model = discovery_info.get(ATTR_UPNP_MODEL_NAME)
+
+        self._name = f"{self._manufacturer} {self._model}"
         self._title = self._model
 
-        # probably access denied
-        if self._id is None:
-            return self.async_abort(reason=RESULT_AUTH_MISSING)
-        if self._id.startswith("uuid:"):
-            self._id = self._id[5:]
+        self.context["title_placeholders"] = {"model": self._model}
+        return await self.async_step_confirm()
 
-        await self.async_set_unique_id(ip_address)
-        self._abort_if_unique_id_configured(
-            {
-                CONF_ID: self._id,
-                CONF_MANUFACTURER: self._manufacturer,
-                CONF_MODEL: self._model,
-            }
-        )
+    async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
+        """Handle a flow initialized by zeroconf discovery."""
+        self._host = discovery_info[CONF_HOST]
+
+        LOGGER.debug("Found Samsung device via zeroconf at %s", self._host)
+
+        await self._async_set_device_unique_id()
+
+        self._mac = discovery_info[ATTR_PROPERTIES].get("deviceid")
+        self._manufacturer = discovery_info[ATTR_PROPERTIES].get("manufacturer")
+        if not self._model:
+            self._model = discovery_info[ATTR_PROPERTIES].get("model")
```

Comment:
```suggestion
        self._model = discovery_info[ATTR_PROPERTIES].get("model")
```

No need to check if its already set this this is the start of the flow

Reply:
I guess its possible that it could be missing from device info?